config.go

```diff
@@ -231,3 +226,33 @@ func GetBrokerConfiguration(config ConfigStruct) BrokerConfiguration {
 func GetMetricsConfiguration(config ConfigStruct) MetricsConfiguration {
 	return config.Metrics
 }
+
+// updateConfigFromClowder updates the current config with the values defined in clowder
+func updateConfigFromClowder(c *ConfigStruct) error {
+	if !clowder.IsClowderEnabled() || clowder.LoadedConfig == nil {
+		fmt.Println("Clowder is disabled")
+		return nil
+	}
+
+	fmt.Println("Clowder is enabled")
+	if clowder.LoadedConfig.Kafka == nil {
+		fmt.Println("No Kafka configuration available in Clowder, using default one")
+	} else {
+		broker := clowder.LoadedConfig.Kafka.Brokers[0]
+		// port can be empty in clowder, so taking it into account
+		if broker.Port != nil {
+			c.Broker.Address = fmt.Sprintf("%s:%d", broker.Hostname, *broker.Port)
+		} else {
+			c.Broker.Address = broker.Hostname
+		}
+
+		// get DB configuration from clowder
+		c.Storage.PGDBName = clowder.LoadedConfig.Database.Name
```

Comment:
IIRC the notification writer uses a different database from the rest of the pipeline. Is this the correct DB?

Reply:
I'm not sure as I'm a "clowder-newbie", but the database configuration retrieved from clowder lib is global to all our namespace or it depends on the configuration defined in the clowder deployment file?

Reply:
I'm afraid that it could return this database https://github.com/RedHatInsights/insights-results-aggregator/blob/master/deploy/clowdapp.yaml#L331 used by the aggregator and defined for the whole pipeline with terraform or something (I don't remember). Or it could return nothing because we are not using `database` definition in the clowdapp.yaml like in the aggregator. Since we want to use this database https://github.com/RedHatInsights/ccx-notification-writer/blob/master/deploy/clowdapp.yaml#L246 we probably don't need to change this configuration.  I can try to test this with ephemeral environments and let you know what happens.